exporting/prometheus/prometheus.c

```diff
@@ -27,19 +27,24 @@ inline int can_send_rrdset(struct instance *instance, RRDSET *st)
     if (unlikely(rrdset_flag_check(st, RRDSET_FLAG_EXPORTING_IGNORE)))
         return 0;
 
-    if (unlikely(!rrdset_flag_check(st, RRDSET_FLAG_EXPORTING_SEND))) {
-        // we have not checked this chart
-        if (simple_pattern_matches(instance->config.charts_pattern, st->id) ||
-            simple_pattern_matches(instance->config.charts_pattern, st->name))
-            rrdset_flag_set(st, RRDSET_FLAG_EXPORTING_SEND);
-        else {
-            rrdset_flag_set(st, RRDSET_FLAG_EXPORTING_IGNORE);
-            debug(
-                D_EXPORTING,
-                "EXPORTING: not sending chart '%s' of host '%s', because it is disabled for exporting.",
-                st->id,
-                host->hostname);
+    if (filter_string) {
+        if (!(simple_pattern_matches(filter, st->id) || simple_pattern_matches(filter, st->name)))
```

Comment:
I think we need to match either id or name (depends on the `send names instead of ids` option), but never both.

Reply:
Done for `prometheus` and `shell` formats. `json` format displays both - `id` and `name`.